Model/Resolver/setRzpPaymentDetailsOnCart.php

```diff
@@ -0,0 +1,202 @@
+     */
+    protected $invoiceService;
+
+    /**
+     * @var \Magento\Framework\DB\Transaction
+     */
+    protected $transaction;
+
+    /**
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface
+     */
+    protected $scopeConfig;
+
+    /**
+     * @var STATUS_PROCESSING
+     */
+    protected const STATUS_PROCESSING = 'processing';
+
+    /**
+     * @param GetCartForUser $getCartForUser
+     * @param SetPaymentMethodOnCartModel $setPaymentMethodOnCart
+     * @param CheckCartCheckoutAllowance $checkCartCheckoutAllowance
+     * @param PaymentMethod $paymentMethod
+     * @param \Magento\Sales\Api\Data\OrderInterface $order
+     * @param \Razorpay\Magento\Model\Config $config
+     * @param \Magento\Sales\Model\Service\InvoiceService $invoiceService
+     * @param \Magento\Framework\DB\Transaction $transaction
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        GetCartForUser $getCartForUser,
+        SetPaymentMethodOnCartModel $setPaymentMethodOnCart,
+        CheckCartCheckoutAllowance $checkCartCheckoutAllowance,
+        PaymentMethod $paymentMethod,
+        \Magento\Sales\Api\Data\OrderInterface $order,
+        \Razorpay\Magento\Model\Config $config,
+        \Magento\Sales\Model\Service\InvoiceService $invoiceService,
+        \Magento\Framework\DB\Transaction $transaction,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+    ) {
+        $this->getCartForUser             = $getCartForUser;
+        $this->setPaymentMethodOnCart     = $setPaymentMethodOnCart;
+        $this->checkCartCheckoutAllowance = $checkCartCheckoutAllowance;
+        $this->rzp                        = $paymentMethod->rzp;
+        $this->order                      = $order;
+        $this->config                     = $config;
+        $this->invoiceService             = $invoiceService;
+        $this->transaction                = $transaction;
+        $this->scopeConfig                = $scopeConfig;
+        $this->_objectManager             = \Magento\Framework\App\ObjectManager::getInstance();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        if (empty($args['input']['order_id'])) {
+            throw new GraphQlInputException(__('Required parameter "order_id" is missing.'));
+        }
+
+        $order_id = $args['input']['order_id'];
+        if (empty($args['input']['rzp_payment_id'])) {
+            throw new GraphQlInputException(__('Required parameter "rzp_payment_id" is missing.'));
+        }
+
+        $rzp_payment_id = $args['input']['rzp_payment_id'];
+        if (empty($args['input']['rzp_order_id'])) {
+            throw new GraphQlInputException(__('Required parameter "rzp_order_id" is missing.'));
+        }
+
+        $rzp_order_id = $args['input']['rzp_order_id'];
```

Comment:
@enishant , we need to use this from our DB not directly from the user input.

Reply:
@ramth05 updated code to fetch rzp_order_id from database